instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/events/consumer/process_message.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      module Events
+        module Consumer
+          # The ProcessMessage module contains the instrumentation for consumer process message event
+          module ProcessMessage
+            extend self
+
+            EVENT_NAME = 'process_message.consumer.kafka'
+            SPAN_NAME = 'process'
+
+            def subscribe!
+              ::ActiveSupport::Notifications.subscribe(EVENT_NAME) do |_name, start, finish, _id, payload|
```

Comment:
This is a really tricky loop to instrument, with the way this is setup we'll generate a span for each processed message but the trace won't propagate further as we're creating the span after the fact so to speak.

Would love to improve this, just unclear what is the way forward here.

The code in question.
https://github.com/zendesk/ruby-kafka/blob/4e63e1ddd2da37f19703ba7a01b13dbc7c8f9a3f/lib/kafka/consumer.rb#L243-L254